scipy/datasets/_utils.py

```diff
@@ -28,6 +67,10 @@ def _clear_cache(datasets, cache_dir=None, method_map=None):
     if datasets is None:
         print(f"Cleaning the cache directory {cache_dir}!")
         shutil.rmtree(cache_dir)
+        if os.path.exists(_CACHE_DIR_FILE):
+            # Remove temporary file containing cache path
+            print(f"Cleaning the tmp file {_CACHE_DIR_FILE}!")
+            os.remove(_CACHE_DIR_FILE)
```

Comment:
Should we also remove the tmp file, when clearing cache? By nature I'd assume yes, since it should reset everything back to defaults.

But one drawback is that, everytime a user runs the test suite, it will also automatically delete the tmp file, losing their custom state if any.